lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java

```diff
@@ -311,7 +311,7 @@ private static boolean isNightMode(Context context) {
    * @see #fromJsonInputStreamSync(InputStream, String, boolean)
    */
   public static LottieTask<LottieComposition> fromJsonInputStream(final InputStream stream, @Nullable final String cacheKey) {
-    return cache(cacheKey, () -> fromJsonInputStreamSync(stream, cacheKey));
+    return cache(cacheKey, () -> fromJsonInputStreamSync(stream, cacheKey), () -> closeQuietly(stream));
```

Comment:
`fromJsonInputStreamSync` closes the stream [here](https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/LottieCompositionFactory.java#L332). Is that not sufficient for your use case?

Reply:
Thank you for reviewing.

Yes, `fromJsonInputStreamSync` doesn't close my stream because I set stream and cacheKey via `setAnimation(InputStream stream, @Nullable String cacheKey)` and hit the cache, but `fromJsonInputStreamSync` doesn't execute when there is cache.

I also can't close my stream manually because the Lottie framework internally closes my stream via `fromJsonInputStreamSync` when there is no cache.